blender-2.5/exporter/osg/osgobject.py

```diff
@@ -28,7 +28,7 @@
 Vector = mathutils.Vector
 FLOATPRE = 5
 CONCAT = lambda s, j="": j.join(str(v) for v in s)
-STRFLT = lambda f: "%%.%df" % FLOATPRE % float(f)
+STRFLT = lambda f: "{}f".format(float(f))
```

Comment:
I see two issues:

* `FLOATPRE` is no longer used but still in the code
* we talked about *not* pre-serializing floating values in string to avoid precision issues by default

You should rewrite `STRFLT` returning the input value if no `FLOATPRE` is set and returning a string if the parameter is set and then let `format` take care of writing float/string data to the file (which you already did).